joi/src/__tests__/joi.ts

```diff
@@ -3,31 +3,67 @@ import { joiResolver } from '..';
 
 const schema = Joi.object({
   username: Joi.string().alphanum().min(3).max(30).required(),
-
-  password: Joi.string().pattern(new RegExp('^[a-zA-Z0-9]{3,30}$')),
-
+  password: Joi.string().pattern(new RegExp('^[a-zA-Z0-9]{3,30}$')).required(),
   repeatPassword: Joi.ref('password'),
-
   accessToken: [Joi.string(), Joi.number()],
-
   birthYear: Joi.number().integer().min(1900).max(2013),
-
   email: Joi.string().email({
     minDomainSegments: 2,
     tlds: { allow: ['com', 'net'] },
   }),
+  tags: Joi.array().items(Joi.string()).required(),
+  enabled: Joi.boolean().required(),
 });
 
+interface Data {
+  username: string;
+  password: string;
+  repeatPassword: string;
+  accessToken?: number | string;
+  birthYear?: number;
+  email?: string;
+  tags: string[];
+  enabled: boolean;
+}
+
 describe('joiResolver', () => {
-  it('should return correct value', async () => {
-    const data = { username: 'abc', birthYear: 1994 };
-    expect(await joiResolver(schema)(data)).toEqual({
-      values: data,
-      errors: {},
-    });
+  it('should return values from joiResolver when validation pass', async () => {
+    const data: Data = {
+      username: 'Doe',
+      password: 'Password123',
+      repeatPassword: 'Password123',
+      birthYear: 2000,
+      email: 'john@doe.com',
+      tags: ['tag1', 'tag2'],
+      enabled: true,
+    };
+
+    const result = await joiResolver(schema)(data);
+
+    expect(result).toEqual({ errors: {}, values: data });
   });
 
-  it('should return errors', async () => {
-    expect(await joiResolver(schema)({})).toMatchSnapshot();
+  it('should return a single error from joiResolver when validation fails', async () => {
+    const data = {
+      password: '___',
+      email: '',
+      birthYear: 'birthYear',
+    };
+
+    const result = await joiResolver(schema)(data);
+
+    expect(result).toMatchSnapshot();
```

Comment:
Minor: (don't have to change as well) maybe be specific with the error? I normally use snapshots for a successful responses.